godot-core/src/obj/oneditor.rs

```diff
@@ -0,0 +1,241 @@
+///         // Will always be valid and must be set via editor – an additional check is being run before ready to make sure that given value can't be null.
+///         let some_variant = self.editor_field.get_meta("SomeName");
+///     }
+/// }
+/// ```
+///
+/// ## Example - user-generated init
+///
+/// ```
+///  use godot::prelude::*;
+///
+/// #[derive(GodotClass)]
+/// #[class(base = Node)]
+/// struct MyClass {
+///     #[export]
+///     required_node: OnEditor<Gd<Node>>,
+///     base: Base<Node>
+/// }
+///
+/// #[godot_api]
+/// impl INode for MyClass {
+///     fn init(base: Base<Node>) -> Self {
+///        Self {
+///            base,
+///            required_node: OnEditor::init(Node::new_alloc()),
+///        }
+///     }
+/// }
+///```
+///
+/// ## Example - Late init
+///
+/// ```
+///  use godot::prelude::*;
+///
+/// #[derive(GodotClass)]
+/// #[class(init, base = Node)]
+/// struct SomeClassThatCanBeInstantiatedInCode {
+///     #[export]
+///     required_node: OnEditor<Gd<Node>>,
+/// }
+///
+/// fn foo(mut this: Gd<Node>) {
+///     let mut my_node_to_add = SomeClassThatCanBeInstantiatedInCode::new_alloc();
+///     // Would cause the panic:
+///     // this.add_child(&my_node_to_add);
+///     // Note: Remember that nodes are manually managed. They will leak memory if not added to tree and/or pruned.
+///     my_node_to_add.bind_mut().required_node = OnEditor::init(Node::new_alloc());
+///     // Will not cause the panic.
+///     this.add_child(&my_node_to_add);
+/// }
+/// ```
+///
+/// # Using `OnEditor<T>` with other GodotTypes
+///
+/// `OnEditor<T>` can be used with other built-ins to provide extra validation logic and making sure that given properties has been set.
+/// Example usage might be checking if entities has been granted proper generated ids.
+///
+/// In such cases the default value which will be deemed invalid **must** be specified with `#[init(val = OnEditor::uninit(...)]`.
+/// Accessing uninitialized value will cause the panic.
+/// To initialize given value simply replace it with `OnEditor::init(…)`.
+///
+/// ## Example - using `OnEditor` with primitives
+///
+/// ```
+///  use godot::prelude::*;
+///
+/// #[derive(GodotClass)]
+/// #[class(init, base = Node)]
+/// struct SomeClassThatCanBeInstantiatedInCode {
+///     #[export]
+///     #[init(val = OnEditor::uninit(42))]
+///     some_primitive: OnEditor<i64>,
+/// }
+///
+/// fn foo(mut this: Gd<Node>) {
+///     let mut my_node_to_add = SomeClassThatCanBeInstantiatedInCode::new_alloc();
+///     // Would cause the panic:
+///     // this.add_child(&my_node_to_add);
+///     my_node_to_add.bind_mut().some_primitive = OnEditor::init(45);
+///     // Will not cause the panic.
+///     this.add_child(&my_node_to_add);
+/// }
+/// ```
+///
+#[doc(
+    alias = "impl<T> export for Gd<T>",
+    alias = "gd_export",
+    alias = "dyn_gd_export",
+    alias = "impl<T, D> export for DynGd<T, D>"
+)]
```

Comment:
I think aliases are only useful for search, and only on exact matches -- and it seems very unlikely that someone enters those keywords into the search bar :thinking: or what was your intention here?

Reply:
the intention was to improve discoverability :thinking: 

I removed aliases altogether, they don't work